src/core/mesh/qgsmeshdataprovidertemporalcapabilities.h

```diff
@@ -0,0 +1,139 @@
+#ifndef QGSMESHDATAPROVIDERTEMPORALCAPABILITIES_H
+#define QGSMESHDATAPROVIDERTEMPORALCAPABILITIES_H
+
+#include "qgsdataprovidertemporalcapabilities.h"
+#include "qgsrange.h"
+#include "qgsmeshdataset.h"
+
+
+/**
+ * \class QgsMeshDataProviderTemporalCapabilities
+ * \ingroup core
+ * Class for handling properties relating to a mesh data provider's temporal capabilities.
+ *
+ * \since QGIS 3.14
+ */
+class CORE_EXPORT QgsMeshDataProviderTemporalCapabilities: public QgsDataProviderTemporalCapabilities
+{
+  public:
+    /**
+     * Constructor for QgsMeshDataProviderTemporalCapabilities
+     */
+    QgsMeshDataProviderTemporalCapabilities();
+
+    /**
+     * Returns the last dataset index that has the time equal or lesser or than \a timeSinceGlobalReference (in milliseconds)from the dataset \a group
+     *
+     * Returns invalide dataset index if the time is less than the first dataset's time
+     */
+    QgsMeshDatasetIndex datasetIndexTimeInMilliseconds( int group, qint64 timeSinceGlobalReference ) const;
+
+    /**
+     * Returns the last dataset index that has the time equal or lesser or than \a timeSinceGlobalReference (in hours)from the dataset \a group
+     *
+     * Returns invalide dataset index if the time is less than the first dataset's time
+     */
+    QgsMeshDatasetIndex datasetIndex( int group, double  timeSinceGlobalReferenceInHours ) const;
+
+    /**
+     * Adds a \a reference date/time from a dataset \a group
+     *
+     * \note must be used only by the mesh data provider
+     */
+    void addGroupReferenceDateTime( int group, const QDateTime &reference ) SIP_SKIP;
+
+    /**
+     * Adds a \a time (in miliseonds) from a dataset contained in \a group
+     *
+     * \note must be used only by the mesh data provider,
```

Comment:
this is class related to the mesh layers (QgsMeshDataProviderTemporalCapabilities) , so maybe note is not relavent, since it couldn't be used for raster/vectors?

Reply:
This comment (as for raster) means that this class is done to be handle by the provider and its usage have to be avoided from other class. This is the same comments as for raster provider capabilities.